### PR TITLE
RavenDB-20339 - Shrading - Tests - Move replication slow tests to use sharded database

### PR DESCRIPTION
**Pull request by lastav5**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-20339

### Additional description

Changing replication's test infrastructure to be generic for both sharded and regular tests.
Migrating all existing tests to sharded and regular modes.

### Type of change

- Tests

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### CODE REVIEW

File: test/Tests.Infrastructure/ReplicationTestBase.cs [hunk 1]
```diff
@@ -39,29 +41,189 @@ public ReplicationTestBase(ITestOutputHelper output) : base(output)
         {
         }
 
-        public class BrokenReplication
+        public class ReplicationInstance : IDisposable
```
Comment (karmeli87) [inline test/Tests.Infrastructure/ReplicationTestBase.cs]: let's move this to a dedicated file

File: test/Tests.Infrastructure/ReplicationTestBase.cs [hunk 2]
```diff
@@ -39,29 +41,189 @@ public ReplicationTestBase(ITestOutputHelper output) : base(output)
...
+            }
+
+            internal static async ValueTask<ReplicationInstance> GetReplicationInstanceAsync(RavenServer server, string databaseName)
+            {
+                DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Stav, DevelopmentHelper.Severity.Normal, "Make this func private when legacy BreakReplication() is removed");
+                return new ReplicationInstance(await server.ServerStore.DatabasesLandlord.TryGetOrCreateResourceStore(databaseName), databaseName);
+            }
+        }
+
+        public class ReplicationManager : ReplicationInstance
```
Comment (karmeli87) [inline test/Tests.Infrastructure/ReplicationTestBase.cs]: let's move this to a dedicated file

**Review by karmeli87 — APPROVED** [review]

File: test/Tests.Infrastructure/ReplicationTestBase.cs (hunk 1 above)
Comment (lastav5) [reply test/Tests.Infrastructure/ReplicationTestBase.cs]: done

File: test/Tests.Infrastructure/ReplicationTestBase.cs (hunk 2 above)
Comment (lastav5) [reply test/Tests.Infrastructure/ReplicationTestBase.cs]: done

**Comment by karmeli87** [comment]
replaced by https://github.com/ravendb/ravendb/pull/16420

**Pull request closed without merging**